merkle/src/tree_impl.rs

```diff
@@ -78,6 +100,64 @@ where
         }
     }
 
+    /// Gets the merkle proof of an item with the 0-based index of the item
+    /// being added, e.g. the first item is index 0
+    pub fn get_proof(
+        &mut self,
+        index: usize,
+    ) -> Option<DefaultMerkleProof<Item, Hash, Hasher, Merger>> {
+        if self.calculate_hash_if_needed(0).is_some() {
+            let height_above = if let Some(fixed_height) = self.mode.fixed_height() {
+                fixed_height
+            } else {
+                self.variable_height
+            };
+            let index_offset = calculate_node_count(height_above);
+            let capacity = self.mode.fixed_height().unwrap_or(self.variable_height) as usize;
+            let mut peer_indices = Vec::with_capacity(capacity);
+            let mut peer_hashes = Vec::with_capacity(capacity);
+            let (item, _) = &self.leafs[index];
```

Comment:
is there a possiblity of panicking on indexing here? does `get_proof` always receives a valid index?

Reply:
Yes, updated comment doc.  This function is mainly for unit tests, we will be constructing proof from graphQL API directly in  the web node